### PR TITLE
Improve floating-point number handling

### PR DESCRIPTION
**Pull request by tylerberry**
This pull request enables warnings for suspicious implicit casts, and fixes all warnings generated by enabling those warnings.

This involved switching from the use of float or double-specific functions (for example, floor vs floorf) to using CGFloat as the floating-point type and <tgmath.h> for the function prototypes. This results in code that works correctly and with the native floating-point width on both 32-bit and 64-bit platforms, and is warning-free!

### CODE REVIEW

File: BFColorPickerPopover/Extensions/NSColor+BFColorPickerPopover.m [hunk 1]
```diff
@@ -44,7 +44,7 @@ - (CGColorRef)copyCGColor
 }
 
 + (NSColor *)randomColor {
-	return [NSColor colorWithCalibratedRed:(float)random()/RAND_MAX green:(float) random()/RAND_MAX blue:(float) random()/RAND_MAX alpha:1.0f];
+	return [NSColor colorWithCalibratedRed:random()/RAND_MAX green:random()/RAND_MAX blue:random()/RAND_MAX alpha:1.0];
```
Comment (DrummerB) [inline BFColorPickerPopover/Extensions/NSColor+BFColorPickerPopover.m]: Are you sure this is ok? random() and RAND_MAX are both integers, but we want floating point division. I can't test this right now.

Comment (tylerberry) [inline BFColorPickerPopover/Extensions/NSColor+BFColorPickerPopover.m]: Good catch! I'll make a fixed pull request shortly.

**Comment by tylerberry** [comment]
Fixed to use an explicit cast to force FP division, which you correctly pointed out was necessary.

I also changed it to use arc4random() instead of random(), which is a "more random" random version. It's documented to return uint32_t, so UINT32_MAX is the correct divisor.

**Comment by DrummerB** [comment]
Alright, merged it. Thanks for the contribution, Tyler!

**Pull request merged**